tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py

```diff
@@ -45,61 +56,130 @@ def create_assets():
         for asset in extra_assets:
             if FILE_NAME_ATTRIBUTE not in asset:
-                print(f"Asset {asset} has no _file_name")
-                continue
+                raise Exception(f"Asset {asset} has no {FILE_NAME_ATTRIBUTE}")
             file_name = asset.pop(FILE_NAME_ATTRIBUTE)
 
             # Find the right folder to create the asset in
             for asset_name, folder in ASSET_FOLDER_MAPPING.items():
-                if asset_name in asset:
-                    if folder == "databases":
-                        # This will fix the URI connection string by setting the right password.
-                        create_superset_db(
-                            asset["database_name"], asset["sqlalchemy_uri"]
-                        )
-                    elif folder == "dashboards":
-                        dashboard_roles = asset.pop("_roles", None)
-                        if dashboard_roles:
-                            roles[asset["uuid"]] = [security_manager.find_role(role) for role in dashboard_roles]
-
-                    path = f"{BASE_DIR}/{folder}/{file_name}"
-                    os.makedirs(os.path.dirname(path), exist_ok=True)
-                    file = open(path, "w")
-                    yaml.dump(asset, file)
-                    file.close()
-                    break
-
-    # Create the zip file and import the assets
-    zip_path = "/app/assets/assets.zip"
-    with ZipFile(zip_path, "w") as zip:
+                if not asset_name in asset:
+                    continue
+
+                write_asset_to_file(asset, asset_name, folder, file_name, roles)
+                break
+
+    create_zip_and_import_assets()
+    update_dashboard_roles(roles)
+
+
+def get_uuid5(base_uuid, name):
+    """Generate an idempotent uuid."""
+    base_uuid = uuid.UUID(base_uuid)
+    base_namespace = uuid.uuid5(base_uuid, "superset")
+    return uuid.uuid5(base_namespace, name)
+
+
+def write_asset_to_file(asset, asset_name, folder, file_name, roles):
+    if folder == "databases":
+        # This will fix the URI connection string by setting the right password.
+        create_superset_db(asset["database_name"], asset["sqlalchemy_uri"])
+
+    asset_translation = TRANSLATIONS.get(asset.get("uuid"), {})
+
+    for language, title in asset_translation.items():
+        updated_asset = generate_asset(asset, asset_name, folder, language, title, roles)
+
+        path = f"{BASE_DIR}/{folder}/{file_name}-{language}.yaml"
+        with open(path, "w") as file:
+            yaml.dump(updated_asset, file)
+
+    dashboard_roles = asset.pop("_roles", None)
+    if dashboard_roles:
+        roles[asset["uuid"]] = [
+            security_manager.find_role(role) for role in dashboard_roles
+        ]
+
+    path = f"{BASE_DIR}/{folder}/{file_name}.yaml"
+    with open(path, "w") as file:
+        yaml.dump(asset, file)
+
+
+def generate_asset(asset, asset_name, folder, language, title, roles):
```

Comment:
where is this being used? I can't find other references

Reply:
On the `write_asset_to_file` function